stock_reserve/models/stock_reserve.py

```diff
@@ -0,0 +1,235 @@
+from odoo import models, fields, api, _
+from odoo.tools.float_utils import float_round
+from odoo.exceptions import UserError
+
+class StockReservation(models.Model):
+    """ Allow to reserve products.
+
+    The fields mandatory for the creation of a reservation are:
+
+    * product_id
+    * product_uom_qty
+    * product_uom
+    * name
+
+    The following fields are required but have default values that you may
+    want to override:
+
+    * company_id
+    * location_id
+    * location_dest_id
+
+    Optionally, you may be interested to define:
+
+    * date_validity  (once passed, the reservation will be released)
+    * note
+    """
+    _name = 'stock.reservation'
+    _description = 'Stock Reservation'
+    _inherits = {'stock.move': 'move_id'}
+
+    move_id = fields.Many2one(
+        'stock.move',
+        'Reservation Move',
+        required=True,
+        readonly=True,
+        ondelete='cascade',
+        index=True)
+    date_validity = fields.Date('Validity Date')
+
+    @api.model
+    def default_get(self, fields_list):
+        """ Fix default values
```

Comment:
IMHO this is quite weird to do such kind of things in default_get for a Model.

In which case default  field attribute does not work ?

Reply:
Sorry, that's the old code of version 8.0.So I don't know much detail about the logic case.